### PR TITLE
feat(hooks-store): add dapp id to hook callData

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

In different places we need a way to display an order hook details (at least a dapp which is created the hook).
To do that we first of all need a registry with all known hook-dapps, here is it ([hookDappsRegistry.json](https://github.com/cowprotocol/cowswap/pull/4920/files#diff-a7a6a3682f59bd901c50ad96c51191a940c65bd728255daef2f4ff676b29dd6c)).

I also added `id` field to each hook-dapp in order to use it for matching hooks with their dapps.
In this PR the hook-dapp id is appended to the hook callData, but we decided to allocated a separate field in the hook data ([PR](https://github.com/cowprotocol/app-data/pull/66)). This will be applied in another PR on top of this.

# To Test

No need to test (please test all recent changes all together when design for hook details is ready)

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/hooksStore/hooks/useInternalHookDapps.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { HookDapp } from '../types/hooks'
+
+export function useInternalHookDapps(isPreHook: boolean): HookDapp[] {
+  const { chainId } = useWalletInfo()
+
+  return useMemo(() => {
+    return ALL_HOOK_DAPPS.filter((dapp) => {
+      if (dapp?.conditions?.supportedNetworks && !dapp.conditions.supportedNetworks.includes(chainId)) return false
+
+      if (isPreHook && dapp?.conditions?.position && dapp.conditions.position !== 'pre') return false
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/hooksStore/hooks/useInternalHookDapps.ts:15]: What it this is a post hook (`isPreHook === false`) and the hook only allows it to be set as pre hook?

File: apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-}
+import { HookDappIframe } from '../../../types/hooks'
 
-type HookDappBaseInfo = Omit<HookDappBase, 'type'>
+type HookDappBaseInfo = Omit<HookDappBase, 'type' | 'conditions'>
 
-type HookDappManifest = HookDappBaseInfo & {
-  conditions?: HookDappConditions
-}
+const MANDATORY_DAPP_FIELDS: (keyof HookDappBaseInfo)[] = ['id', 'name', 'image', 'version', 'website']
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx:15]: Thinking about this now.
What happens when there's a duplicate id or name?
Do we have a registry somewhere?

**Review by alfetopito — APPROVED** [review]
Small comments/questions

File: apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx (hunk 2 above)
Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx:15]: Ah, I found the registry file.

File: apps/cowswap-frontend/src/modules/hooksStore/hooks/useInternalHookDapps.ts (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooksStore/hooks/useInternalHookDapps.ts]: Thank you!
Fixed that: https://github.com/cowprotocol/cowswap/pull/4920/commits/1cf305ffd80658aac0146dee2a152458e879a082

**Pull request merged**